### PR TITLE
Error message translation

### PR DESCRIPTION
**Pull request by JeroenVinke**
So I played around a bit with the translation bit (since I need it in a project), and I got it working so I thought I'd open a PR as it may help with https://github.com/aurelia/validatejs/issues/27. I only tried it with the fluent setup, not with the decorators.  

Basically it allows you to provide translation services at two levels, a [rule specific level](https://github.com/JeroenVinke/validatejs/blob/i18n-pr/sample/src/i18n.js#L15) and a [global level](https://github.com/JeroenVinke/validatejs/blob/i18n-pr/sample/src/main.js#L19-L23). 

image

### CODE REVIEW

File: sample/src/main.js [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+          loadPath: '/locales/{{lng}}/{{ns}}.json'
+        },
+        lng: 'nl'
+      });
+    })
+    .plugin('aurelia-validatejs', config => {
+      let i18n = aurelia.container.get(I18N);
+      config.useTranslation(error => {
+        let translation = i18n.tr(error.rule, error);
+        return translation !== error.rule ? translation : undefined;
```
Comment (JeroenVinke) [inline sample/src/main.js]: this is pretty ugly as you can't ask i18n if there is a translation for a key. When there isn't a translation for a key, it just returns the key

File: sample/src/i18n.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  model;
+  errors = [];
+  subscriber;
+  constructor() {
+    this.model = new Model();
+    this.validator = new Validator(this)
+     .ensure('model.firstName')
+       .required()
+       .length({minimum: 5, maximum: 20})
+       .errorHandler(error => `${error.propertyName} is wrong`)
```
Comment (JeroenVinke) [inline sample/src/i18n.js]: I'm not sure it `errorHandler` is the nicest name

File: src/index.js [hunk 3]
```diff
@@ -7,9 +7,16 @@ import {ValidationReporter} from 'aurelia-validation';
...
-export function configure(config) {
-  config.container.registerHandler(Validator, ValidateJSValidator);
-  config.container.registerHandler(ValidationReporter, ValidateJSReporter);
-  config.globalResources('./validate-binding-behavior');
+export function configure(aurelia, config) {
+  aurelia.container.registerHandler(Validator, ValidateJSValidator);
+  aurelia.container.registerHandler(ValidationReporter, ValidateJSReporter);
+  aurelia.globalResources('./validate-binding-behavior');
+
+  ValidationConfig.prototype.configBuilder = new ConfigBuilder();
```
Comment (JeroenVinke) [inline src/index.js]: setting things on the prototype isn't the nicest solution, but we need to get this ConfigBuilder instance from `validation-config.js` that doesn't use DI. Is it OK to use `Container.instance.get()` there?

**Comment by EisenbergEffect** [comment]
@jdanyow I'm not sure if we can accept this PR but can you have a look soon? We definitely want to make sure there's a way to integrate i18n with validation.

**Comment by EisenbergEffect** [comment]
Closing this PR for now. There are some discussions in issues regarding i18n support. Please provide feedback there. Thanks!

**Pull request closed without merging**